dotcom-rendering/src/model/enhance-H2s.ts

```diff
@@ -1,15 +1,54 @@
+import { JSDOM } from 'jsdom';
+
+const extractText = (element: SubheadingBlockElement): string => {
+	const frag = JSDOM.fragment(element.html);
+	if (!frag.firstElementChild) return '';
+	return frag.textContent?.trim() ?? '';
+};
+
+/**
+ * We need the ids to be unique so we can navigate to them.
+ * This function checks if the slug already exists and if it does it adds the count to the end of the slug.
+ */
+export const getUnique = (slug: string, array: string[]): string => {
+	if (array.includes(slug)) {
+		const occurenceCount = array.filter(
+			(currentItem) => currentItem == slug,
+		).length;
+		return `${slug}-${occurenceCount}`;
+	}
+	return slug;
+};
+
+const slugify = (text: string) => {
+	return text
+		.normalize('NFKD') // The normalize() using NFKD method returns the Unicode Normalization Form of a given string.
+		.toLowerCase() // Convert the string to lowercase letters
+		.trim() // Remove whitespace from both sides of a string
+		.replace(/\s+/g, '-') // Replace spaces with "-"
+		.replace(/[^\w-]+/g, '') // Remove all non-word chars
+		.replace(/--+/g, '-'); // Replace multiple "-" with single "-"
+};
+
 const enhance = (elements: CAPIElement[]): CAPIElement[] => {
+	const slugifiedIds: string[] = [];
 	const enhanced: CAPIElement[] = [];
 	elements.forEach((element) => {
-		// If this element is a subheading, insert the element ID as its ID
+		// If this element is a subheading, insert a humanised slug of the text as its ID
 		if (
 			element._type ===
 			'model.dotcomrendering.pageElements.SubheadingBlockElement'
 		) {
+			const text = extractText(element);
+			const slug = text ? slugify(text) : element.elementId;
```

Comment:
Can `slugify` ever return a falsey value?

Reply:
`''` is falsy I believe? I think it's possible for it to return an empty string if the processing filters out all characters. For example, if the input is a space: `' '`.

Reply:
Yep, I think you're right here. I'll put a check in place to make sure we always get a viable id here